src/explorer/LeetCodeTreeDataProvider.ts

```diff
@@ -13,6 +13,8 @@ import { LeetCodeNode } from "./LeetCodeNode";
 
 export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCodeNode> {
 
+    private allProblems: Map<string, IProblem>; // store reference of all problems.
```

Comment:
Any reason we separate `allProblems` with `treeData`?

Reply:
* `allProblems` is an id-IProblem dictionary primarily for fast lookup, whose structure is not compatible with `treeData`;
* `allProblems` serves as a buffer which holds all the problems' ownership, where elements in `treeData` is just references to the problems;
* with regard to `this.onDidChangeTreeDataEvent.fire(problem)` review below, we may need another lookup dictionary for fast retrieving relative nodes. `allProblems` seems more similar to it rather than `treeData`.

Reply:
Now `allProblems` is decoupled with `treeData`, and will be grouped with `allTreeNodes` as an object pool in next PR.